src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -91,10 +91,12 @@ class ActionPlanService(
 
   fun approveActionPlan(id: UUID, user: AuthUser): ActionPlan {
     val actionPlan = getActionPlan(id)
```

Comment:
One more thing: Do you think we should do a NOOP if actionPlan == referral.approvedActionPlan?

I suppose in the case of the user submitting it twice by mistake or double clicking...

Reply:
To avoid mishandling race conditions like @andrewburrowsmoj mentioned we would need to know the ID of the action plan that is being approved

Then we could reject it if
- already approved
- not the 'latest' non-approved

Reply:
Thinking about it some more -- the method is already idempotent since no new sessions would be created. So perhaps this check is unnecessary..

However the operation is not atomic - so it's theoretically possible for duplicate requests to happen where the second request starts when the first request is still in flight.
The possibility of this happening is very low - except in the case of a double submit with mouse button - we normally add a javascript blocker on the front-end to prevent this.

Personally I think making things atomic is the only way to have 100% certainty - but also realise that we don't do that everywhere.

Reply:
I think this is a business rule we need to enforce -- PPs shouldn't be able to approve action plans if there's a newer submitted version already (plus add in the re-approval checks)

Reply:
for the not latest non-approved check. Should this be the latest non-approved submitted or created action plan